### PR TITLE
fix(ui): Preserve whitespace in alert messages

### PR DESCRIPTION
**Pull request by rhcarvalho**
And add a full stop to the end of the messages.

image

### CODE REVIEW

File: src/sentry/static/sentry/app/components/alert.tsx [hunk 1]
```diff
@@ -95,7 +95,10 @@ const Alert = styled(
           {typeof icon === 'string' ? <InlineSvg src={icon} size={iconSize!} /> : icon}
         </IconWrapper>
       )}
-      <StyledTextBlock>{children}</StyledTextBlock>
+      <StyledTextBlock>
+        {/* wrap with <span> to preserve inline white space in children */}
```
Comment (rhcarvalho) [inline src/sentry/static/sentry/app/components/alert.tsx]: Idea taken from https://stackoverflow.com/questions/39325039/css-flex-box-last-space-removed#comment94925741_39334882

**Comment by rhcarvalho** [comment]
@iProgramStuff FYI

**Review by untitaker — APPROVED** [review]
hm. something in the css must have changed because this used to be fine

**Review by iProgramStuff — COMMENTED** [review]
Lgtm. Thanks for doing this. Travis isn't very happy though.

I approve of the addition of the period too :)

**Comment by rhcarvalho** [comment]
> something in the css must have changed because this used to be fine

Yes, in #17237 we can see the screenshots, it was okay.

I used `git bisect` and found that the change was introduced in https://github.com/getsentry/sentry/pull/17840, most certainly an accident after changing the box model to `flex`, merged to master just 2 days ago.

Funny thing is there are bad changes visible in Percy, but you need to be paranoid to details to see it, e.g. https://percy.io/getsentry/sentry/builds/4636410/view/272627326?browser=firefox&mode=head&width=1280

**Comment by rhcarvalho** [comment]
The current patch is still not great... thanks to [the Percy screenshot I posted above](https://percy.io/getsentry/sentry/builds/4636410/view/272627326?browser=firefox&mode=head&width=1280), from the time we introduced the flex box, I went to look at other alerts:

image

**Review by priscilawebdev — APPROVED** [review]

**Comment by priscilawebdev** [comment]
@rhcarvalho Please check Percy...maybe you need to rebase master

**Comment by rhcarvalho** [comment]
> Please check Percy

It needs to be checked carefully, there are indeed render changes to the alert boxes. Hopefully we introduce no new regression.

**Comment by rhcarvalho** [comment]
For the travis failure, I need to update some snapshots to the DOM to include the new `<span>`.

**Comment by rhcarvalho** [comment]
> I need to update some snapshots to the DOM to include the new `<span>`.

Snapshots updated.

**Review by iProgramStuff — APPROVED** [review]

**Comment by rhcarvalho** [comment]
Just for reference, this was the first complete Percy build: https://percy.io/getsentry/sentry/builds/4669239

Looking through it I found stuff broken in even uglier ways

image

**Comment by rhcarvalho** [comment]
All good, and by reviewing Percy I found https://github.com/getsentry/sentry/issues/17939.

Merging.

File: src/sentry/static/sentry/app/components/alert.tsx [hunk 2]
```diff
@@ -95,7 +95,10 @@ const Alert = styled(
...
+        <span>{children}</span>
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/alert.tsx]: @rhcarvalho This is going to make it hard if we need to customize the contents of the Alert.

Comment (rhcarvalho) [reply src/sentry/static/sentry/app/components/alert.tsx]: The alternative was fixing each individual alert box.

This is one of the boxes we had in prod for a couple of days:

image

My intention here was just to bring back the correct look of alerts (and I did check all of the ones we capture in Percy).

If this is really a problem, we need to revisit how we are aligning the icon, see #17840 where we introduced the bug when we changed the box model to `flex`. Most likely we don't need to make `StyledTextBlock` be `display: flex`.

**Pull request merged**